### PR TITLE
baseline files updated

### PR DESCRIPTION
**Pull request by aminaramoon**
Input.* and variable.* have been updated, the core code is intact, I have added an inner class called TypeDetector in variable class, and only 4 new input commands in input file to create Type detectors.

TypeDetecor class is used to dynamically assign the type of bonds, angles, dihedrals or improper dihedrals in fix_bond_create class. The current version of the "bond create" allow only for a static type for angle, dihedrals and improper dihedrals. however, the type  of coordinates created can vary based on the type of atom that are involved in formation of them. TypeDetector takes care of that. 

I will submit the update version of "fix_bond_create" class that utilizes the added features in the input, varialbe classes in another pull request as you asked yesterday. The code that I have added has backward compatibility and I have not changed any lines of codes in these files expect the added features.

### CODE REVIEW

**Comment by akohlmey** [comment]
First off, the addition to the Variable class has no relation at all to variables, so why add it there? and why the addition to the Input class instead of adding command styles?

But even more important, I don't see any reason, why this functionality needs to be added to the baseline. It could be added to fix bond/create straight away. What is the reason to do it differently?
Rather than having those individual commands (which are badly named, by the way. the names bond type, angle type, and so on have already a different meaning inside LAMMPS, so using these commands in this way is very confusing to other users).

In short, i don't see a chance, that steve will accept these changes.

**Comment by aminaramoon** [comment]
The reason that it is in input class is that it will make using it much simpler !

for instance:

angle_type  m_angles 1 2 1 1 2 1 2 2 3 3 1 3 1 3 1 4 2 1 3 5 * * * 6
dihedral_type m_dihedrals 1 2 1 2 1 * * * * 2

fix   1 all bond/create 10 1 3 10 1 iparam 2 1 jparam 3 3 atype m_angles dtype m_dihedrals

user has to use something like : 

fix   1 all bond/create 10 1 3 10 1 iparam 2 1 jparam 3 3 atype "1 2 1 1 2 1 2 2 3 3 1 3 1 3 1 4 2 1 3 5 * * * 6" dtype "1 2 1 2 1 * * * * 2"

which is very confusing

**Comment by aminaramoon** [comment]
The second reason is, an updated create_atom class that utilizes the TypeDetectors to randomly generates polymer chains inside the box, or region.

**Comment by akohlmey** [comment]
you can use ${var} variable style expansion with string style variable to simplify the input. or write the modified bond/create fix so that it explicitly requires a string (or index or similar) style variable.
The latter will give you essentially the same user interface, but you get to keep all code in the fix.

**Comment by aminaramoon** [comment]
Okay, I will move the code in fix class. 

Thanks again.

**Comment by akohlmey** [comment]
as for the modified create_atoms class, i think you'd be better of to not modify create_atoms, but have a separate "create_chains" class. You can collect the type detection related code in a separate TypeDetector class, and then both, your "create_chains" and your "fix bond/create" commands, can access an instance of that class to get access to the shared functionality. This would also have the benefit, that you don't need to expose any additional data structures (e.g. the priority list) to the global headers. A simple forward declaration of TypeDetector and holding a pointer to it will do.

I've just implemented something *very* similar as an extension to the load-balancing commands. Both, the balance command, and fix balance make use of functionality provided by classes, that are derived from class called Imbalance (in imbalance.h).

**Comment by aminaramoon** [comment]
Sure, I will do. 
Thanks for the suggestion

**Comment by akohlmey** [comment]
Here is another thought: If your modifications to fix bond/create are exclusively or mostly added features, you may implement your modifications as a derived class fix bond/create/xlink rather than modifying fix bond/create directly. small modifications to fix bond/create (e.g. to change visibility of class members) would be acceptable. that would keep all modifications in your package folder and make more modular and significantly more likely to be included "as is".
